packages/truffle-hdwallet-provider/src/index.js

```diff
@@ -116,6 +116,8 @@ function HDWalletProvider(
 
   this.engine.addProvider(new FiltersSubprovider());
   if (typeof provider === "string") {
+    Web3.providers.HttpProvider.prototype.sendAsync =
+      Web3.providers.HttpProvider.prototype.send;
```

Comment:
Could we just get rid of `sendAsync` instead of patching it in?

Reply:
If this is necessary for reasons specified in one of the four linked issues, I apologize for being lazy.

Reply:
It's necessary for at least one of the linked issues, plus now `web3-provider-engine` throws an err if a provider (or maybe subProvider?) doesn't have that method.